### PR TITLE
Issue #201 User request ticket instead of global ticket before commit

### PR DESCRIPTION
**Pull request by gregw**
This is not yet ready to merge as it needs testing.  PR created now to facilitate discussion on #201

### CODE REVIEW

File: appengine-managed-runtime/src/main/java/com/google/apphosting/vmruntime/VmApiProxyEnvironment.java [hunk 1]
```diff
@@ -526,8 +528,13 @@ public String getServer() {
     return server;
   }
 
+  public void clearTicket() {
+    ticket=null;
+  }
+  
   public String getTicket() {
-    return ticket;
+    String t=ticket;
```
Comment (ludoch) [inline appengine-managed-runtime/src/main/java/com/google/apphosting/vmruntime/VmApiProxyEnvironment.java]: nit: space before and after =

File: appengine-managed-runtime/src/main/java/com/google/apphosting/vmruntime/VmApiProxyEnvironment.java [hunk 2]
```diff
@@ -526,8 +528,13 @@ public String getServer() {
...
+    return (t!=null&&!t.isEmpty())?t:globalTicket;
   }
```
Comment (ludoch) [inline appengine-managed-runtime/src/main/java/com/google/apphosting/vmruntime/VmApiProxyEnvironment.java]: space before and after operators

**Comment by gregw** [comment]
I signed it!

**Comment by ludoch** [comment]
Ready for merge or doing more testing?

**Comment by gregw** [comment]
Ludo,

I think the changes in this PR are good.   It is certainly better than just using global ticket, but we should probably open an issue somewhere to clarify how long the request ticket is valid for, so we can keep using it as long as possible.

**Comment by gregw** [comment]
... so yes good to merge

**Pull request merged**